src/app/shared/data-access/event.service.ts

```diff
@@ -65,10 +137,79 @@ export class EventService {
       ),
     );
 
-  readonly events = toSignal(this.events$);
-  readonly eventForThisRound = toSignal(this.eventForThisRound$);
   readonly eventForNextRound = toSignal(this.eventForNextRound$);
 
+  readonly events = toSignal(this.events$);
+
+  readonly eventTeams$: Observable<EventTeamModel[] | null> = this.events$.pipe(
+    map((events) => events?.map((ev) => ev.id)),
+    whenNotNull((eventIds) =>
+      realtimeUpdatesFromTable(
+        this.supabase,
+        Table.EventTeam,
+        `event_id=in.(${eventIds})`,
+      ),
+    ),
+    shareReplay(1),
+  );
+
+  readonly eventTeams = toSignal(this.eventTeams$);
+
+  readonly eventTeamsWithParticipantInfo: Signal<
+    EventTeamWithParticipantInfo[] | undefined
+  > = computed(() =>
+    this.eventTeams()?.map((eventTeam) => ({
+      ...eventTeam,
+      participants: this.eventParticipantsWithPlayerInfo()?.filter(
+        (eventParticipant) => eventParticipant.team_id === eventTeam.id,
+      ),
+    })),
```

Comment:
To go along with our theme of "undefined means loading" and "null means loaded, but no data", we can modify this Signal to handle both of those cases:

```typescript
readonly eventTeamsWithParticipantInfo: Signal<EventTeamWithParticipantInfo[] | null | undefined
> = computed(() => {
  const eventTeams = this.eventTeams();
  const eventParticipantsWithPlayerInfo = this.eventParticipantsWithPlayerInfo();
  
  // return null or undefined, depending on what the value is
  if (!eventTeams) return eventTeams;
  if (!eventParticipantsWithPlayerInfo) return eventParticipantsWithPlayerInfo;
  
  
  return this.eventTeams.map((eventTeam) => ({
    ...eventTeam,
    participants: eventParticipantsWithPlayerInfo.filter(
      (eventParticipant) => eventParticipant.team_id === eventTeam.id,
    ),
  }));
});
```

Reply:
Agreed, I'll add this when I put the Signal into the Event Service.